packages/orm/src/changes.ts

```diff
@@ -58,14 +67,30 @@ export function newChangesProxy<T extends Entity>(entity: T): Changes<T> {
   return new Proxy(entity, {
     get(target, p: PropertyKey): FieldStatus<any> | ManyToOneFieldStatus<any> | (keyof OptsOf<T>)[] {
       if (p === "fields") {
-        return (
+        const fieldsChanged = (
           entity.isNewEntity
             ? // Cloning sometimes leaves unset keys in data as undefined, so drop them
               Object.entries(getInstanceData(entity).data)
                 .filter(([, value]) => value !== undefined)
                 .map(([key]) => key)
             : Object.keys(getInstanceData(entity).originalData)
         ) as (keyof OptsOf<T>)[];
+
+        // scan the join rows to get if the relation has any rows
+        const m2mFieldsChanged: (keyof RelationsOf<T>)[] = [];
+        const emApi = getEmInternalApi(entity.em);
+        // we will report changes only on the many to many relations for now
+        const m2mFields = Object.values(getMetadata(entity).allFields).filter((f) => f.kind === "m2m");
+        for (const field of m2mFields) {
+          const m2m = entity[field.fieldName as keyof T] as ManyToManyCollection<any, any>;
+          const joinRow: any = emApi.joinRows(m2m);
```

Comment:
The `: any` is not a great hack; also I think just checking `joinRow.rows.length > 0` is going to over-report changes, like it's legal for `joinRow.rows` to be populated, and not actually have any changes queued up.

Can you remove this `any` and isntead do something like:

```
if (emAPi.joinRows(m2m).hasChanges) {
  . ..
}
```

Where `hasChanges` is a method/property in `JoinRows` that does a copy/paste/reuse of `toTodo` which specifically scans for added/deletes m2m rows.

Reply:
Adding 👍